src/platform_impl/windows/event_loop.rs

```diff
@@ -2073,122 +2128,232 @@ unsafe extern "system" fn thread_event_target_callback<T: 'static>(
+                                Event::MouseEvent(mouse_id.into(), MouseEvent::Removed)
+                            }
+                            DeviceId::Keyboard(keyboard_id) => {
+                                Event::KeyboardEvent(keyboard_id.into(), KeyboardEvent::Removed)
+                            }
+                            DeviceId::Hid(hid_id) => {
+                                Event::HidEvent(hid_id.into(), HidEvent::Removed)
+                            }
+                            DeviceId::Gamepad(gamepad_handle, _) => {
+                                Event::GamepadEvent(gamepad_handle.into(), GamepadEvent::Removed)
+                            }
+                        };
+                        subclass_input.send_event(event);
+                    }
+                }
+                _ => unreachable!(),
+            }
 
             0
         }
 
         winuser::WM_INPUT => {
-            use crate::event::{
-                DeviceEvent::{Button, Key, Motion, MouseMotion, MouseWheel},
-                ElementState::{Pressed, Released},
-                MouseScrollDelta::LineDelta,
-            };
-
-            if let Some(data) = raw_input::get_raw_input_data(lparam as _) {
-                let device_id = wrap_device_id(data.header.hDevice as _);
+            use crate::event::ElementState::{Pressed, Released};
 
-                if data.header.dwType == winuser::RIM_TYPEMOUSE {
-                    let mouse = data.data.mouse();
+            match get_raw_input_data(lparam as _) {
+                Some(RawInputData::Mouse {
+                    device_handle,
+                    raw_mouse,
+                }) => {
+                    let mouse_handle = MouseId(device_handle).into();
 
-                    if util::has_flag(mouse.usFlags, winuser::MOUSE_MOVE_RELATIVE) {
-                        let x = mouse.lLastX as f64;
-                        let y = mouse.lLastY as f64;
+                    if util::has_flag(raw_mouse.usFlags, winuser::MOUSE_MOVE_ABSOLUTE) {
+                        let x = raw_mouse.lLastX as f64;
+                        let y = raw_mouse.lLastY as f64;
 
-                        if x != 0.0 {
-                            subclass_input.send_event(Event::DeviceEvent {
-                                device_id,
-                                event: Motion { axis: 0, value: x },
-                            });
-                        }
-
-                        if y != 0.0 {
-                            subclass_input.send_event(Event::DeviceEvent {
-                                device_id,
-                                event: Motion { axis: 1, value: y },
-                            });
+                        if x != 0.0 || y != 0.0 {
+                            subclass_input.send_event(Event::MouseEvent(
+                                mouse_handle,
+                                MouseEvent::MovedAbsolute(PhysicalPosition { x, y }),
```

Comment:
> Do you know in what coordinate system other platforms report absolute position?

I have no idea.

Reply:
So, I tested this with my Wacom tablet, since it's the only device I have that can report absolute position, and the coordinates it spits out are not normalized at all. It just gives me regular "pixel coordinates".

Reply:
I plugged my old Wacom tablet as well and it reports values in the range of [0; 64k] following the [Windows docs](https://docs.microsoft.com/en-us/windows/win32/api/winuser/ns-winuser-rawmouse#remarks).
I had to disable Ink as it otherwise broke Raw Input for me (not reporting values at all only on entering/exiting a Window)
```
[src\platform_impl\windows\event_loop.rs:2226] (x, y, raw_mouse.usFlags) = (
    61224.0,
    62378.0,
    MOUSE_MOVE_ABSOLUTE | MOUSE_VIRTUAL_DESKTOP,
)
[src\platform_impl\windows\event_loop.rs:2226] (x, y, raw_mouse.usFlags) = (
    60098.0,
    61524.0,
    MOUSE_MOVE_ABSOLUTE | MOUSE_VIRTUAL_DESKTOP,
)
```